editor/plugins/script_text_editor.cpp

```diff
@@ -1732,39 +1765,65 @@ void ScriptTextEditor::drop_data_fw(const Point2 &p_point, const Variant &p_data
 	int row = pos.y;
 	int col = pos.x;
 
+	const bool use_type = EDITOR_GET("text_editor/completion/add_type_hints");
+	const bool drop_modifier_pressed = Input::get_singleton()->is_key_pressed(Key::CMD_OR_CTRL);
+
+	const String &line = te->get_line(row);
+	const bool is_empty_line = line.is_empty() || te->get_first_non_whitespace_column(row) == line.length();
+
 	if (d.has("type") && String(d["type"]) == "resource") {
 		te->remove_secondary_carets();
-		Ref<Resource> res = d["resource"];
-		if (!res.is_valid()) {
+		Ref<Resource> resource = d["resource"];
+		if (!resource.is_valid()) {
 			return;
 		}
 
-		if (res->get_path().is_resource_file()) {
+		const String &path = resource->get_path();
+		if (path.is_empty() || path.ends_with("::")) {
+			String warning = TTR("The resource does not have a valid path because it has not been saved.\nPlease save the scene or resource that contains this resource and try again.");
+			EditorToaster::get_singleton()->popup_str(warning, EditorToaster::SEVERITY_ERROR);
+			return;
+		}
+		if (path.is_resource_file()) {
 			EditorNode::get_singleton()->show_warning(TTR("Only resources from filesystem can be dropped."));
```

Comment:
Not really related, but I think this check does not make sense. You can remove it.